library/common/http/header_utility.cc

```diff
@@ -9,11 +10,16 @@ namespace Http {
 namespace Utility {
 
 RequestHeaderMapPtr toRequestHeaders(envoy_headers headers) {
-  RequestHeaderMapPtr transformed_headers = RequestHeaderMapImpl::create();
+  auto transformed_headers = RequestHeaderMapImpl::create();
+  transformed_headers->setFormatter(
+      std::make_unique<
+          Extensions::Http::HeaderFormatters::PreserveCase::PreserveCaseHeaderFormatter>());
+  Envoy::Http::StatefulHeaderKeyFormatter& formatter = transformed_headers->formatter().value();
   for (envoy_map_size_t i = 0; i < headers.length; i++) {
-    transformed_headers->addCopy(
-        LowerCaseString(Data::Utility::copyToString(headers.entries[i].key)),
-        Data::Utility::copyToString(headers.entries[i].value));
+    std::string key = Data::Utility::copyToString(headers.entries[i].key);
+    formatter.processKey(key);
```

Comment:
So is the idea here that a copy of the non-lower-cased header name (key) is being saved in the formatter where it can be retrieved down in toBridgeHeaders? If so, perhaps add a comment to this effect. When I read `processKey()` I expected that the key was being transformed but I think that's not quite the case?